### PR TITLE
feature: add Google Authentication screen

### PR DESCRIPTION
**Pull request by robertaleG**
Simple google auth screen conforms with Figma and redirects to Overview screen. Once user fetching with email is done, we can change it to redirect to profile creation screen or overview.

### CODE REVIEW

File: app/src/androidTest/java/com/github/onlynotesswent/ui/authentication/LoginTest.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
+package com.github.onlynotesswent.ui.authentication
+
```
Comment (Roshan-y) [inline app/src/androidTest/java/com/github/onlynotesswent/ui/authentication/LoginTest.kt:2]: The comment from line 3 to 9 is unnecessary and should be removed

**Review by Roshan-y — CHANGES_REQUESTED** [review]
Code written well, remove the unnecessary comments (I added as a comment in review) and add more tests to pass 80% code coverage

**Review by Roshan-y — CHANGES_REQUESTED** [review]
Code well written, need to delete Authentication.kt as it is not used anymore

File: app/src/main/java/com/github/onlynotesswent/MainActivity.kt [hunk 2]
```diff
@@ -43,7 +43,7 @@ fun OnlyNotesApp() {
         startDestination = Screen.AUTH,
         route = Route.AUTH,
     ) {
-      composable(Screen.AUTH) { AuthenticationScreen(navigationActions) }
```
[the remark this answers is not in the document]
Comment (robertaleG) [reply app/src/main/java/com/github/onlynotesswent/MainActivity.kt:46]: Changes were addressed in latest commit.

**Review by Roshan-y — APPROVED** [review]

**Pull request merged**